docs/source/concepts/CachingLayers.rst

```diff
@@ -0,0 +1,233 @@
+.. _CachingLayers:
+
+==============================================
+Caching Layers - SpectrumInfo and DetectorInfo
+==============================================
+
+.. contents::
+  :local:
+
+Introduction
+------------
+
+There are two caching layers, ``SpectrumInfo`` and ``DetectorInfo``, that are going to be introduced to Mantid as part of the work towards Instrument 2.0. Eventually these classes will cache commonly accessed information about spectra or detectors, namely masking, monitor flags, L1, L2, 2-theta and position, leading to improved performance and cleaner code.
+
```

Comment:
Maybe add a short note, similar to this:

A spectrum corresponds to (a group of) one or more detectors. Most algorithm work with spectra and thus `SpectrumInfo` would be used. Some algorithms work on a lower level (with individual detectors) and thus `DetectorInfo` would be used.